node/runtime/src/lib.rs

```diff
@@ -391,16 +391,18 @@ impl Runtime {
                 match receipt.body {
                     ReceiptBody::NewCall(ref mut call) => {
                         if call.callback.is_some() {
-                            return Err("receipt already has callback".to_string());
+                            return Err("don't return original promise that already has a callback".to_string());
                         } else {
                             call.callback = Some(callback_info.clone());
                         }
                     }
-                    _ => unreachable!("receipt body is not new call")
+                    _ => unreachable!("receipt body is not a new call")
                 }
                 None
             }
-            _ => return Err("return data is a non-callback promise".to_string())
+            ReturnData::Promise(PromiseId::Joiner(_)) => {
+                return Err("don't return a joined promise".to_string())
```

Comment:
it's not immediately clear what does "joined" mean, maybe use something like "don't return a joined (using `promise_and` or `Promise.all`) promise"?